webapp/src/slash_commands.tsx

```diff
@@ -0,0 +1,222 @@
+import {defineMessage} from 'react-intl';
+
+import {CommandArgs} from '@mattermost/types/integrations';
+
+import {getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentUserId, getUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {getChannel as getChannelAction} from 'mattermost-redux/actions/channels';
+
+import {
+    startCallRecording,
+    stopCallRecording,
+    displayGenericErrorModal,
+} from 'src/actions';
+import {SHOW_END_CALL_MODAL} from 'src/action_types';
+import {DisabledCallsErr} from 'src/constants';
+
+import {Store} from './types/mattermost-webapp';
+import {logDebug} from './log';
+import {sendDesktopEvent, shouldRenderDesktopWidget} from './utils';
+import {
+    connectedChannelID,
+    voiceConnectedUsersInChannel,
+    voiceChannelCallOwnerID,
+    voiceChannelCallHostID,
+    callRecording,
+} from './selectors';
+
+type joinCallFn = (channelId: string, teamId: string, title?: string, rootId?: string) => void;
+
+export default async function slashCommandsHandler(store: Store, joinCall: joinCallFn, message: string, args: CommandArgs) {
+    const fullCmd = message.trim();
+    const fields = fullCmd.split(/\s+/);
+    if (fields.length < 2) {
+        return {message, args};
+    }
+
+    const rootCmd = fields[0];
+    const subCmd = fields[1];
+
+    if (rootCmd !== '/call') {
+        return {message, args};
+    }
+
+    const connectedID = connectedChannelID(store.getState());
+
+    switch (subCmd) {
+    case 'join':
+    case 'start':
+        if (subCmd === 'start') {
+            if (voiceConnectedUsersInChannel(store.getState(), args.channel_id).length > 0) {
+                store.dispatch(displayGenericErrorModal(
+                    defineMessage({defaultMessage: 'Unable to start call'}),
+                    defineMessage({defaultMessage: 'A call is already ongoing in the channel.'}),
+                ));
+                return {};
+            }
+        }
+        if (!connectedID) {
+            let title = '';
+            if (fields.length > 2) {
+                title = fields.slice(2).join(' ');
+            }
+
+            let channel = getChannel(store.getState(), args.channel_id);
+            if (!channel) {
+                await store.dispatch(getChannelAction(args.channel_id));
+                channel = getChannel(store.getState(), args.channel_id);
```

Comment:
This can be simplified I think:
```suggestion
                const {data: channelData} = await store.dispatch(getChannelAction(args.channel_id));
                channel = channelData
```

Reply:
I didn't know that, nice!